src/filesystem-upload.js

```diff
@@ -190,25 +190,28 @@ export default class FileSystemUpload extends DirectBinaryUpload {
                         errors,
                         totalSize
                     } = await walkDirectory(currPath, getMaxFileCount(options), isDeep);
-                    // base sub-item remote paths on current path's parent so that they end up in
-                    // the correct directory
-                    const currPathParent = Path.dirname(currPath);
+                    const itemManager = new FileSystemUploadItemManager(options, currPath, !isDeep);
                     if (isDeep) {
                         // directories only need to be included for deep uploads
-                        allDirectories.push({ remoteUrl: getItemRemoteUrl(options, '', currPath), path: currPath });
-                        allDirectories = allDirectories.concat(directories.map(dirItem => {
-                            const { path: dirPath } = dirItem;
-                            return { remoteUrl: getItemRemoteUrl(options, currPathParent, dirPath), path: dirPath };
-                        }));
+                        allDirectories.push(await itemManager.getDirectory(currPath));
+                        const subDirectories = [];
+                        for (let d = 0; d < directories.length; d += 1) {
```

Comment:
It's safe here to use the increment operator...
```suggestion
                        for (let d = 0; d < directories.length; d++) {
```